server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -781,7 +781,8 @@
         Pair<List<UserVmJoinVO>, Integer> result = searchForUserVMsInternal(cmd);
         ListResponse<UserVmResponse> response = new ListResponse<UserVmResponse>();
         ResponseView respView = ResponseView.Restricted;
-        if (cmd instanceof ListVMsCmdByAdmin) {
+        Account caller = CallContext.current().getCallingAccount();
+        if (_accountMgr.isAdmin(caller.getId())) {
```

Comment:
@davidjumani  it should be isRootAdmin instead of isAdmin

Reply:
Does it detect ListVMsCmd as ListVMsCmdByAdmin for a regular admin ?

Reply:
@davidjumani yes, root admin. Check if `isAdmin` checks for only root admin, or would it pass for domain admin. The fix needed here is to check if caller is a root admin.

Reply:
Nevermind, I checked and made that fix.